tests/test_data.py

```diff
@@ -3228,3 +3229,101 @@ async def test_retrieve_optional_field_no_default(init_dataclasses_and_load_sche
 
     report = await data.Report.get_by_id(report.id)
     assert report.returncode is None
+
+
+@pytest.fixture
+def simple_query_builder():
+    return SimpleQueryBuilder(
+        select_clause="SELECT *",
+        from_clause="FROM table",
+        filter_statements=["column = $1"],
+        values=["value"],
+        limit=10,
+        backward_paging=False,
+        prelude="prelude_query AS (SELECT * FROM prelude_table)",
+        prelude_extra=[
+            "extra_prelude_1 AS (SELECT * FROM extra_table1)",
+            "extra_prelude_2 AS (SELECT * FROM extra_table2)",
+        ],
+    )
+
+
+@pytest.fixture
+def prelude_query_builder():
+    return SimpleQueryBuilder(
+        select_clause="SELECT *",
+        from_clause="FROM prelude_table",
+        filter_statements=["prelude_column = $1"],
+        values=["prelude_value"],
+    )
+
+
+def test_prelude_extra_in_simple_query_builder(simple_query_builder):
+    query, _ = simple_query_builder.build()
+    assert (
+        """WITH prelude_query AS (SELECT * FROM prelude_table),
+extra_prelude_1 AS (SELECT * FROM extra_table1),
+extra_prelude_2 AS (SELECT * FROM extra_table2)
+SELECT *"""
+        in query
+    )
+
+
+def test_query_building_in_prelude_based_filtering_query_builder(prelude_query_builder):
+    extra_prelude_builder = SimpleQueryBuilder(
+        select_clause="SELECT *",
+        from_clause="FROM extra_prelude_table",
+        filter_statements=["extra_prelude_column = $2"],
+        values=["extra_prelude_value"],
+    )
+
+    prelude_based_builder = PreludeBasedFilteringQueryBuilder(
+        prelude_query_builder=prelude_query_builder,
+        prelude_query_builder_extra=[extra_prelude_builder],
+        select_clause="SELECT mt.*",
+        from_clause="""
+        FROM main_table mt
+        JOIN prelude p ON mt.some_column = p.some_related_column
+        JOIN extra_prelude_1 ep ON mt.another_column = ep.another_related_column""",
+    )
+
+    query, values = prelude_based_builder.build()
+    assert (
+        """WITH prelude AS (
+SELECT *
+FROM prelude_table
+WHERE prelude_column = $1),
+extra_prelude_1 AS (
+SELECT *
+FROM extra_prelude_table
+WHERE extra_prelude_column = $2)
+SELECT mt.*
+
+        FROM main_table mt
+        JOIN prelude p ON mt.some_column = p.some_related_column
+        JOIN extra_prelude_1 ep ON mt.another_column = ep.another_related_column
+"""
+        in query
```

Comment:
Why use `in` here? This way we cannot be sure whether query contains more than the given string.